frontend/src/app/api/BaseApi.ts

```diff
@@ -1,3 +1,9 @@
+// This server-only package is recommended by Next.js to ensure code is only run on the server.
+// It provides a build-time error if client-side code attempts to invoke the code here.
+// Since we're pulling in an API Auth Token here, this should be server only
+// https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#keeping-server-only-code-out-of-the-client-environment
+import "server-only";
+
```

Comment:
Regarding the difference between server-only and use server:

`use server` is more for indicating a function is a server action.